service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseControllerTest.java

```diff
@@ -716,6 +721,119 @@ void shouldThrowAssersionErrorWhenCaseIdIsInvalid() {
         }
     }
 
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    @Nested
+    class Dfpl684 {
+        private final String migrationId = "DFPL-684";
+        private final long validCaseId = 1642001990437030L;
+        private final long invalidCaseId = 1626258358022000L;
+
+        @Test
+        void shouldPerformMigrationWhenNameMatches() {
+            List<UUID> uuidsToBeDeleted = List.of(UUID.fromString("afae84df-1337-4aa5-90ff-4938dbeb241c"),
+                UUID.fromString("f55cd3ab-cb71-4eeb-b786-076eb8728f7c"));
+
+            List<Element<Respondent>> respondents = List.of(element(Respondent.builder()
+                    .party(RespondentParty.builder()
+                        .firstName("Respondent")
+                        .lastName("One")
+                        .build())
+                    .build()),
+                element(Respondent.builder()
+                    .party(RespondentParty.builder()
+                        .firstName("Respondent")
+                        .lastName("Two")
+                        .build())
+                    .build()),
+                element(uuidsToBeDeleted.get(0), Respondent.builder()
+                    .party(RespondentParty.builder()
+                        .firstName("Respondent")
+                        .lastName("Three")
+                        .build())
+                    .build()),
+                element(Respondent.builder()
+                    .party(RespondentParty.builder()
+                        .firstName("Respondent")
+                        .lastName("Four")
+                        .build())
+                    .build()),
+                element(uuidsToBeDeleted.get(1), Respondent.builder()
+                    .party(RespondentParty.builder()
+                        .firstName("Respondent")
+                        .lastName("Five")
+                        .build())
+                    .build()));
+
+            CaseData caseData = CaseData.builder()
+                .id(validCaseId)
+                .state(State.SUBMITTED)
+                .respondents1(respondents)
+                .respondentStatements(List.of(element(RespondentStatement.builder()
+                    .respondentId(uuidsToBeDeleted.get(0))
+                    .build()),
+                    element(RespondentStatement.builder()
+                        .respondentId(uuidsToBeDeleted.get(1))
+                        .build())))
```

Comment:
I think it's all fine but would it be good just for confirmation to have a respondent statement from one of the respondents we are keeping to make sure it sticks around?

Reply:
yeah, sure! added a statement element which has to be kept.